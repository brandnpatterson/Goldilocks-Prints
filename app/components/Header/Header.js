import React from 'react'
import Navigation from '../Navigation/Navigation'
import './Header.sass'

const logoHeader = 'public/images/goldilocksheader.gif'
const logoBear = 'public/images/goldilocksbear.gif'
const styles = {
  transition: 'all 0.3s'
}

export default class Header extends React.Component {
  constructor () {
    super()
    this.state = {
      transform: false,
      height: 200,
      top: 40,
      logo: logoHeader,
      opacity: 0,
      paddingTop: 45,
      scale: 1
    }
    this.onScroll = this.onScroll.bind(this)
  }
  onScroll () {
    if (document.body.scrollTop > 1) {
      this.setState({
        height: 70,
        logo: logoBear,
        opacity: 1,
        paddingTop: 10,
        top: -13,
        scale: 0.5
      })
    } else {
      this.setState({
        height: 200,
        logo: logoHeader,
        opacity: 0,
        paddingTop: 45,
        top: 40,
        scale: 1
      })
    }
  }
  componentWillMount () {
    window.addEventListener('scroll', this.onScroll)
  }
  render () {
    return (
      <div className='header'>
        <div className='header-background'
          style={{
            ...styles,
            height: this.state.height,
            opacity: this.state.opacity
          }}
        />
        <img
          className='logo'
          style={{
            ...styles,
            top: this.state.top,
            transform: 'scale(' + this.state.scale + ')'
          }}
          src={this.state.logo}
          alt='Goldilocks Bear'
        />
        <div
          style={{
            ...styles,
            paddingTop: this.state.paddingTop
          }}>
          <Navigation />
        </div>
      </div>
    )
  }
}
