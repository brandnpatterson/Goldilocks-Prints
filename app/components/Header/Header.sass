@import ../../index.sass

.header
  width: 80%
  li
    a
      color: $dark
  .header-background
    background: $white
    box-shadow: 3px 0 5px #888
    position: fixed
    width: 100vw
    z-index: 1
  .routes-container
    margin-top: 250px
  .nav-links
    transition: all 0.6s ease-in
    ul
      +mediaQuery($small)
        display: none
      li
        &:nth-child(5)
          position: fixed
          background-color: $gold
          border: 2px solid black
          border-radius: 10px
          margin: 2%
          right: -160px
          padding: 20px
          bottom: 20px
      .shop-now-slide
        animation: shop-now-slide 1s
        animation-fill-mode: forwards
        backface-visibility: hidden
        perspective: 1000
      @keyframes shop-now-slide
        0%
          right: -160px
        100%
          right: 30px
  .nav-links-responsive
    padding: 0
    ul
      display: flex
      flex-direction: row
      flex-wrap: wrap
      margin-top: 100px
      z-index: 2
      li
        border: 2px solid black
        border-radius: 10px
        margin: 1%
        flex: 1 1 20%
        text-align: center
        a
          display: block
          padding: 35px 0
          &:hover
            opacity: 0.5
  .logo
    height: 100px
    padding-left: 33.5px
    position: fixed
    z-index: 1
    +mediaQuery($small)
      margin-left: -30px
  ul
    right: 40px
    position: fixed
    z-index: 1
  li
    padding: 0 20px 0 20px
  .slide-enter
    height: 200px
    opacity: 1
  .slide-enter-active
    height: 70px
    opacity: 0
    transition: opacity 300ms
  .slide-leave
    height: 70px
    opacity: 0
  .slide-leave-active
    height: 200px
    opacity: 1
    transition: opacity 300ms

// shrink-navbar
@keyframes shrink-navbar
  0%
    height: 150px
  100%
    height: 85px

.shrink-navbar
  animation: shrink-navbar 0.5s
  animation-fill-mode: forwards
  backface-visibility: hidden
  perspective: 1000

// grow-navbar
@keyframes grow-navbar
  0%
    height: 85px
  100%
    height: 150px

.grow-navbar
  animation: grow-navbar 0.5s
  animation-fill-mode: forwards
  backface-visibility: hidden
  perspective: 1000

// shrink slide
@keyframes shrink-slide
  0%
    height: 100px
    padding-top: 40px
  100%
    height: 60px
    padding-top: 16px

.shrink-slide
  animation: shrink-slide 0.5s
  animation-fill-mode: forwards
  backface-visibility: hidden
  perspective: 1000

// grow-slide
@keyframes grow-slide
  0%
    height: 60px
    padding-top: 16px
  100%
    height: 100px
    padding-top: 40px

.grow-slide
  animation: grow-slide 0.5s
  animation-fill-mode: forwards
  backface-visibility: hidden
  perspective: 1000
