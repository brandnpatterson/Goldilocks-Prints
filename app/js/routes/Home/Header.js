import React, { Component } from 'react'

import Quote from './Quote'
const imageArr = [ 'public/images/landing1.jpeg', 'public/images/landing2.jpeg', 'public/images/landing1.jpeg' ]

export default class Header extends Component {
  constructor () {
    super()
    this.state = {
      image: imageArr[0]
    }
  }

  componentDidMount () {
    let imagePosition = 0
    setInterval(() => {
      if (imageArr.length - 1 > imagePosition) {
        this.setState({
          image: imageArr[imagePosition]
        })
        imagePosition++
      } else {
        this.setState({
          image: imageArr[imagePosition]
        })
        imagePosition = 0
      }
    }, 4000)
  }

  render () {
    const headerStyle = {
      width: '100vw'
    }
    return (
      <div>
        <img style={headerStyle} src={this.state.image} alt='header images slideshow' />
        <Quote />
      </div>
    )
  }
}
