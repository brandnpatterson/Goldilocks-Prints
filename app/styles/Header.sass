@import './basics/index'
@import './Header/index'

.header
  width: 80%

  li
    a
      color: $dark

  .header-background
    background: $white
    box-shadow: 3px 0 5px #888
    height: 150px
    position: fixed
    width: 100vw
    z-index: 1

  .routes-container
    margin-top: 250px

  .nav-links
    transition: all 0.6s ease-in

    ul
      @include mediaQuery($iPhone)
        display: none

      li
        &:nth-child(5)
          position: fixed
          background-color: $gold
          border: 2px solid black
          border-radius: 10px
          margin: 2%
          right: -150px
          padding: 20px
          bottom: 20px

      .shop-now-slide
        animation: shop-now-slide 1s
        animation-fill-mode: forwards
        backface-visibility: hidden
        perspective: 1000

      @keyframes shop-now-slide
        0%
          right: -150px
        100%
          right: 30px

  .nav-links-visible
    padding: 0

    ul
      display: flex
      flex-direction: row
      flex-wrap: wrap
      margin-top: 100px
      z-index: 2

      li
        border: 2px solid black
        border-radius: 10px
        margin: 1%
        flex: 1 1 20%
        text-align: center

        a
          display: block
          padding: 35px 0

          &:hover
            opacity: 0.5

  .logo
    height: 100px
    padding-left: 33.5px
    position: fixed
    z-index: 1
    @include mediaQuery($iPhone)
      margin-left: -30px

  ul
    right: 40px
    position: fixed
    z-index: 1

  li
    padding: 0 20px 0 20px

#practice li
    position: relative

/* Starting State of Animation */
.fade-enter
    right: 100px

/* End State of Animation */
.fade-enter-active
    right: 0px
    transition: .5s ease-in all
