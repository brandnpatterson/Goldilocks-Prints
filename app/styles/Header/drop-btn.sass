@import '../basics/mixins'

.drop-btn
  cursor: pointer
  display: none
  position: fixed
  padding-top: 40px
  right: 20px
  z-index: 2
  @include mediaQuery($iPhone)
    display: inline-block

  div
    width: 35px
    background-color: $dark
    border-radius: 25px
    height: 5px
    margin: 6px 0

.bar1,
.bar2,
.bar3
  width: 35px
  height: 5px
  margin: 6px 0
  transition: all 0.4s ease-out

.drop-btn-transform
  .bar1
      transform: rotate(-45deg) translate(-9px, 6px)

  .bar2
      opacity: 0

  .bar3
      transform: rotate(45deg) translate(-9px, -6px)
