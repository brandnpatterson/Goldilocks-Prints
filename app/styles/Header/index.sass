@import './drop-btn'
@import './keyframes'
@import './shop-now'
