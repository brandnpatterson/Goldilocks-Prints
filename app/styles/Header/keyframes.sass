// navbar
@keyframes shrink-navbar
  0%
    height: 150px
  100%
    height: 85px

@keyframes grow-navbar
  0%
    height: 85px
  100%
    height: 150px

// shrink & grow navbar
.shrink-navbar
  animation: shrink-navbar 0.5s
  animation-fill-mode: forwards
  backface-visibility: hidden
  perspective: 1000

.grow-navbar
  animation: grow-navbar 0.5s
  animation-fill-mode: forwards
  backface-visibility: hidden
  perspective: 1000


// slide
@keyframes shrink-slide
  0%
    height: 100px
    padding-top: 40px
  100%
    height: 60px
    padding-top: 16px

@keyframes grow-slide
  0%
    height: 60px
    padding-top: 16px
  100%
    height: 100px
    padding-top: 40px

// shrink & grow slide
.shrink-slide
  animation: shrink-slide 0.5s
  animation-fill-mode: forwards
  backface-visibility: hidden
  perspective: 1000

.grow-slide
  animation: grow-slide 0.5s
  animation-fill-mode: forwards
  backface-visibility: hidden
  perspective: 1000
