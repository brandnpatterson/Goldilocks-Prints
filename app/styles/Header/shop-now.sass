.shop-now
  padding-top: 200px

  ul li a
    background: $gold
    border: 2px solid $dark
    border-radius: 5px
    padding: 20px
    // position: fixed
    // bottom: 5%
    // right: 5%
    // z-index: 1

  &:hover
    opacity: 0.5

  &.left
    right: 20%
