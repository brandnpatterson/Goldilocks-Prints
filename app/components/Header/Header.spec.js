import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Header from './Header'

test('It should render the Header component as expected', () => {
  const component = shallow(<Header />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
