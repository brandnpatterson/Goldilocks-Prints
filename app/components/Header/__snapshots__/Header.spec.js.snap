// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render the Header component as expected 1`] = `
<div
  className="header"
>
  <div
    className="header-background"
    style={
      Object {
        "height": 200,
        "opacity": 0,
        "transition": "all 0.3s ease-out",
      }
    }
  />
  <div
    className="drop-btn "
    onClick={[Function]}
    style={
      Object {
        "paddingTop": 45,
        "transition": "all 0.3s ease-out",
      }
    }
  >
    <div
      className="bar1"
    />
    <div
      className="bar2"
    />
    <div
      className="bar3"
    />
  </div>
  <img
    alt="Goldilocks Bear"
    className="logo"
    src="public/images/goldilocksheader.gif"
    style={
      Object {
        "top": 40,
        "transform": "scale(1)",
        "transition": "all 0.3s ease-out",
      }
    }
  />
  <div
    className="nav-links"
    onClick={[Function]}
    style={
      Object {
        "paddingTop": 45,
        "transition": "all 0.3s ease-out",
      }
    }
  >
    <Links />
  </div>
</div>
`;
